From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bjarne Koll <git@lynxplay.dev>
Date: Fri, 29 Aug 2025 12:21:43 +0200
Subject: [PATCH] Overload AsyncCatcher for log-only catches

Useful for purely notifying async catches and early returns when
simply skipping logic is less destructive than throwing an exception.

diff --git a/src/main/java/org/spigotmc/AsyncCatcher.java b/src/main/java/org/spigotmc/AsyncCatcher.java
index 7e7a0ed944961af5ab7a49bc659da2862d2e2c82..defd834191991efd4713739935bb0d7b513d9b2e 100644
--- a/src/main/java/org/spigotmc/AsyncCatcher.java
+++ b/src/main/java/org/spigotmc/AsyncCatcher.java
@@ -10,4 +10,13 @@ public class AsyncCatcher {
             throw new IllegalStateException("Asynchronous " + reason + "!");
         }
     }
+    // paperbin start - non throwing overload
+    public static boolean catchOpNonThrowing(String reason) {
+        if (!ca.spottedleaf.moonrise.common.util.TickThread.isTickThread()) { // Paper - chunk system
+            MinecraftServer.LOGGER.error("Thread {} failed main thread check: {}", Thread.currentThread().getName(), reason, new Throwable()); // Paper
+            return true;
+        }
+        return false;
+    }
+    // paperbin end - non throwing overload
 }
